src/Microsoft.AspNetCore.Authentication/AuthenticationToken.cs

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Http.Authentication;
+
+namespace Microsoft.AspNetCore.Authentication
+{
+    public class AuthenticationToken
+    {
+        public string Name { get; set; }
+        public string Value { get; set; }
+
+        private static string TokenNamesKey = ".TokenNames";
+
+        public static void StoreTokens(AuthenticationProperties properties, IEnumerable<AuthenticationToken> tokens)
```

Comment:
It's a little odd to have these as static here. I'd expect them to be extension methods.